pcsx2/Patch.cpp

```diff
@@ -114,188 +107,116 @@ static int PatchTableExecute(const ParsedAssignmentString& set, const PatchTextT
-		name.MakeUpper();
-		if (name.Find(upperGameCRC) == 0 && name.Find(L".PNACH") + 6u == name.Length())
-		{
-			PatchesCon->WriteLn(Color_Green, L"Loading patch '%s' from archive '%s'",
-								WX_STR(entry->GetName()), WX_STR(patchesArchiveFilename));
-			wxTextInputStream pnach(zip);
-			while (!zip.Eof())
-			{
-				inifile_processString(pnach.ReadLine());
-			}
-		}
-	}
-	return Patch.size() - before;
+	PatchesCon->WriteLn(Color_Green, "Loading patch '%s' from archive.", pnach_filename.c_str());
+	return LoadPatchesFromString(pnach_data.value());
 }
 
 
 // This routine loads patches from *.pnach files
 // Returns number of patches loaded
 // Note: does not reset previously loaded patches (use ForgetLoadedPatches() for that)
-int LoadPatchesFromDir(const wxString& name, const wxDirName& folderName, const wxString& friendlyName)
+int LoadPatchesFromDir(const std::string& crc, const wxDirName& folder, const char* friendly_name, bool show_error_when_missing)
 {
-	int loaded = 0;
-	int numberFoundPatchFiles;
+	if (!folder.Exists())
+	{
+		Console.WriteLn(Color_Red, "The %s folder ('%s') is inaccessible. Skipping...", friendly_name, folder.ToUTF8().data());
+		return 0;
+	}
 
-	wxString filespec = name + L"*.pnach";
-	loaded += _LoadPatchFiles(folderName, filespec, friendlyName, numberFoundPatchFiles);
+	FileSystem::FindResultsArray files;
+	FileSystem::FindFiles(folder.ToUTF8(), StringUtil::StdStringFromFormat("%s*.pnach", crc.c_str()).c_str(),
+		FILESYSTEM_FIND_FILES | FILESYSTEM_FIND_HIDDEN_FILES, &files);
 
-	if (folderName.ToString().IsSameAs(EmuFolders::Cheats.ToString()) && numberFoundPatchFiles == 0)
+	if (show_error_when_missing && files.empty())
 	{
-		wxString pathName = Path::Combine(folderName, name.Upper() + L".pnach");
-		PatchesCon->WriteLn(Color_Gray, L"Not found %s file: %s", WX_STR(friendlyName), WX_STR(pathName));
+		PatchesCon->WriteLn(Color_Gray, "Not found %s file: %s" FS_OSPATH_SEPARATOR_STR "%s.pnach",
+			friendly_name, folder.ToUTF8().data(), crc.c_str());
 	}
 
-	PatchesCon->WriteLn((loaded ? Color_Green : Color_Gray), L"Overall %d %s loaded", loaded, WX_STR(friendlyName));
-	return loaded;
-}
+	int total_loaded = 0;
 
-static u32 StrToU32(const wxString& str, int base = 10)
-{
-	unsigned long l;
-	str.ToULong(&l, base);
-	return l;
-}
+	for (const FILESYSTEM_FIND_DATA& fd : files)
+	{
+		const std::string_view name(FileSystem::GetFileNameFromPath(fd.FileName));
+		PatchesCon->WriteLn(Color_Green, "Found %s file: '%.*s'", friendly_name, static_cast<int>(name.size()), name.data());
 
-static u64 StrToU64(const wxString& str, int base = 10)
-{
-	wxULongLong_t l;
-	str.ToULongLong(&l, base);
-	return l;
+		const std::optional<std::string> pnach_data(FileSystem::ReadFileToString(fd.FileName.c_str()));
+		if (!pnach_data.has_value())
+			continue;
+
+		const int loaded = LoadPatchesFromString(pnach_data.value());
+		PatchesCon->WriteLn((loaded ? Color_Green : Color_Gray), "Loaded %d %s from '%s'.",
```

Comment:
```suggestion
		PatchesCon->WriteLn((loaded ? Color_Green : Color_Gray), "Loaded %d %s from '%.*s'.",
```

Wrong format string explodes debug builds rn